acceptance-tests/src/testFixtures/java/tech/pegasys/teku/test/acceptance/dsl/TekuValidatorNode.java

```diff
@@ -222,9 +268,34 @@ public TekuValidatorNode.Config withInteropValidators(
       return this;
     }
 
+    public TekuValidatorNode.Config withSentryNodes(final SentryNodesConfig sentryNodesConfig) {
+      final File sentryNodesConfigFile;
+      try {
+        sentryNodesConfigFile = File.createTempFile("sentry-node-config", ".json");
+        sentryNodesConfigFile.deleteOnExit();
+
+        try (FileWriter fw = new FileWriter(sentryNodesConfigFile, StandardCharsets.UTF_8)) {
+          fw.write(sentryNodesConfig.toJson(JSON_PROVIDER));
+        }
+      } catch (IOException e) {
+        throw new RuntimeException("Error creating sentry nodes configuration file", e);
+      }
+      configFileMap.put(sentryNodesConfigFile, SENTRY_NODE_CONFIG_FILE_PATH);
+
+      configMap.put("Xsentry-config-file", SENTRY_NODE_CONFIG_FILE_PATH);
+      isUsingSentryNodeConfig = true;
+
+      return this;
+    }
+
     public void writeConfigFile() throws Exception {
       final File configFile = File.createTempFile("config", ".yaml");
       configFile.deleteOnExit();
+
+      if (isUsingSentryNodeConfig) {
+        configMap.remove("beacon-node-api-endpoint");
```

Comment:
does this reduce functionality available to the dsl? just wondering if future AT's won't understand why some api functions don't work, might be worth a log message...

Reply:
Basically, one can't work with the other. If you are using sentry nodes, you cannot set a beacon node api endpoint. However, your duties provider sentry node will be used as your beacon node endpoint.

Reply:
For reference: https://github.com/ConsenSys/teku/blob/master/teku/src/main/java/tech/pegasys/teku/cli/options/ValidatorClientOptions.java#L105-L125

Reply:
as long as dsl functions like getting data via rest api still work, I'm ok with that :)

Reply:
It will work exactly the same way :)